src/index.ts

```diff
@@ -256,13 +256,11 @@ const mkMatchXW =
   <C extends B[keyof B]>(x: A): C => {
     const tag = x[tagKey] as Tag<A>
 
-    const g = fs[tag]
     // eslint-disable-next-line functional/no-conditional-statement, @typescript-eslint/no-unsafe-return
-    if (g !== undefined) return g as C
+    if (tag in fs) return fs[tag] as C
```

Comment:
Perhaps we should use `hasOwnProperty` to exclude properties on the prototype as well?

Reply:
Good call will change this.

Reply:
afd9a761922d43cc3642dcad5cf13ad0aa33744a